packages/versioning/src/versioning.ts

```diff
@@ -58,13 +58,24 @@ export function $removed(context: DecoratorContext, t: Type, v: EnumMember) {
   }
   program.stateMap(removedOnKey).set(t, version);
 }
+
+interface RenamedFrom {
+  version: Version;
+  oldName: string;
+}
+
 export function $renamedFrom(context: DecoratorContext, t: Type, v: EnumMember, oldName: string) {
   const { program } = context;
   const version = checkIsVersion(context.program, v, context.getArgumentTarget(0)!);
   if (!version) {
     return;
   }
-  const record = { v: version, oldName: oldName };
+
+  // retrieve statemap to update or create a new one
+  const record = getRenamedFrom(program, t) ?? [];
+  record.push({ version: version, oldName: oldName });
+  // ensure that records are stored in ascending order
+  (record as Array<RenamedFrom>).sort((a, b) => a.version.index - b.version.index);
```

Comment:
```suggestion
 record.sort((a, b) => a.version.index - b.version.index);
```
I don't think you need that `as` anymore